### PR TITLE
optimize: Only AT mode try to get channel with other app

### PR DESCRIPTION
**Pull request by Bughue**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅰ. Describe what this PR did
tcc模式下，resourceid如果不能保证全局唯一，有可能导致路由错误（TC调用了错误的RM）.
这个pr判断branchtype，只有AT模式才允许tryOtherApp

### Ⅱ. Does this pull request fix one issue?

fixes https://github.com/seata/seata/issues/5130

### CODE REVIEW

File: server/src/main/java/io/seata/server/coordinator/AbstractCore.java [hunk 1]
```diff
@@ -196,11 +197,16 @@ public BranchStatus branchRollback(GlobalSession globalSession, BranchSession br
...
     protected BranchStatus branchRollbackSend(BranchRollbackRequest request, GlobalSession globalSession,
                                               BranchSession branchSession) throws IOException, TimeoutException {
+
         BranchRollbackResponse response = (BranchRollbackResponse) remotingServer.sendSyncRequest(
-                branchSession.getResourceId(), branchSession.getClientId(), request);
+            branchSession.getResourceId(), branchSession.getClientId(), request, tryOtherApp(branchSession));
         return response.getBranchStatus();
     }
 
+    protected boolean tryOtherApp(BranchSession branchSession) {
```
Comment (a364176773) [inline server/src/main/java/io/seata/server/coordinator/AbstractCore.java:206]: 这块应该写在branchsession内部，isAT()

**Review by a364176773 — APPROVED** [review]
LGTM

**Review by l81893521 — APPROVED** [review]
LGTM

**Review by caohdgege — APPROVED** [review]
LGTM

**Pull request merged**